#! /bin/sh /usr/share/dpatch/dpatch-run
## fix_via_sentby_proto.dpatch by  <mikma@users.sourceforge.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Don't include transport protocol in via_sentby used in transaction
## DP: matching. Since CANCEL for an INVITE may not be sent using same
## DP: transport.

@DPATCH@
diff -urNad yxa-1.0+20090326~/src/sipheader.erl yxa-1.0+20090326/src/sipheader.erl
--- yxa-1.0+20090326~/src/sipheader.erl	2009-07-16 22:57:34.000000000 +0200
+++ yxa-1.0+20090326/src/sipheader.erl	2009-07-16 22:57:58.000000000 +0200
@@ -815,11 +815,10 @@
 
 %%--------------------------------------------------------------------
 %% @spec    (Via) ->
-%%            {Proto, Host, Port}
+%%            {Host, Port}
 %%
 %%            Via = #via{}
 %%
-%%            Proto = string()
 %%            Host  = string()
 %%            Port  = integer()
 %%
@@ -827,7 +826,7 @@
 %% @end
 %%--------------------------------------------------------------------
 via_sentby(Via) when is_record(Via, via) ->
-    {Via#via.proto, Via#via.host, Via#via.port}.
+    {Via#via.host, Via#via.port}.
 
 %%--------------------------------------------------------------------
 %% @spec    (Request) ->
@@ -1595,7 +1594,7 @@
 
     autotest:mark(?LINE, "get_server_transaction_id/1 - 1.2"),
     %% check result
-    {"z9hG4bK-really-unique", {"SIP/2.0/TLS", "sip.example.org", 5061}, "INVITE"} = Invite1Id,
+    {"z9hG4bK-really-unique", {"sip.example.org", 5061}, "INVITE"} = Invite1Id,
 
     autotest:mark(?LINE, "get_server_transaction_id/1 - 2"),
     %% make an ACK for an imagined 3xx-6xx response with to-tag "t-123"
@@ -1695,7 +1694,7 @@
     %% test via_sentby(Via)
     %%--------------------------------------------------------------------
     autotest:mark(?LINE, "via_sentby/1 - 1"),
-    {"proto", "host", 1234} = via_sentby(#via{proto="proto", host="host", port=1234}),
+    {"host", 1234} = via_sentby(#via{proto="proto", host="host", port=1234}),
 
 
     %% test via_print(Via)
